src/scene/vcPOI.cpp

```diff
@@ -1337,6 +1341,23 @@ void vcPOI::GenerateLineFillPolygon(vcState *pProgramState)
     vcPolygonModel_CreateFromRawVertexData(&m_pPolyModel, pVerts, numPoints, vcP3N3UV2VertexLayout, (int)(udLengthOf(vcP3N3UV2VertexLayout)));
 
     m_pPolyModel->modelOffset = udDouble4x4::translation(m_line.pPoints[0]);
+    
+    // Calculate area
+    m_meshArea = 0.0;
+    for (int i = 0; i < trianglePointList.size(); i += 3)
+    {
+      udFloat3 e1 = {
+        pVerts[i + 1].position.x - pVerts[i].position.x,
+        pVerts[i + 1].position.y - pVerts[i].position.y,
+        pVerts[i + 1].position.z - pVerts[i].position.z };
+      udFloat3 e2 = {
+        pVerts[i + 2].position.x - pVerts[i].position.x,
+        pVerts[i + 2].position.y - pVerts[i].position.y,
+        pVerts[i + 2].position.z - pVerts[i].position.z };
+      udFloat3 e3 = udCross3<float>(e1, e2);
+
+      m_meshArea += (double)(udMag3(e3) / 2.0f);
```

Comment:
Again, you should do all calculations in doubles. Then convert to floats if the renderer needs floats.